### PR TITLE
Add creating schedules triggered by schedule and names for schedules

### PR DESCRIPTION
**Pull request by Tantalor93**
Issue #374 
 - When creating schedule, you can either specify cron expression or different schedule, which triggers execution of created schedule (you cannot specify both)

-  You can also optionally set schedule name (which differs from default name of schedule set to name of executable)

### CODE REVIEW

File: src/main/java/com/gooddata/dataload/processes/Schedule.java [hunk 1]
```diff
@@ -69,6 +70,27 @@ public Schedule(final DataloadProcess process, final String executable, final St
         this.links = Collections.emptyMap();
     }
 
+    /**
+     * Creates schedule, which is triggered by execution of different schedule
+     * @param process process to create schedule for
+     * @param executable executable to be scheduled for execution
+     * @param triggerSchedule schedule, which will trigger created schedule
+     */
+    public Schedule(final DataloadProcess process, final String executable, final Schedule triggerSchedule) {
```
Comment (martiner) [inline src/main/java/com/gooddata/dataload/processes/Schedule.java]: do we really need a new constructor, isn't the setter enough?

Comment (Tantalor93) [inline src/main/java/com/gooddata/dataload/processes/Schedule.java]: yes, we need a way to create schedule without having to specify cron expression. If both cron and triggerScheduleId is specified (= not null), you will not be able to create/update schedule.

File: src/main/java/com/gooddata/dataload/processes/Schedule.java [hunk 2]
```diff
@@ -69,6 +71,27 @@ public Schedule(final DataloadProcess process, final String executable, final St
...
+
+        this.type = MSETL_TYPE;
+        this.state = ScheduleState.ENABLED.name();
+        this.params = new HashMap<>();
+        this.params.put(PROCESS_ID, process.getId());
+        process.validateExecutable(executable);
+        this.nextExecutionTime = null;
+        this.consecutiveFailedExecutionCount = 0;
+        this.params.put(EXECUTABLE, executable);
+        this.triggerScheduleId = notEmpty(triggerSchedule.getId(), "trigger schedule ID cannot be null!");
```
Comment (martiner) [inline src/main/java/com/gooddata/dataload/processes/Schedule.java]: please validate triggerSchedule for nullability

File: src/main/java/com/gooddata/dataload/processes/Schedule.java (hunk 1 above)
Comment (martiner) [inline src/main/java/com/gooddata/dataload/processes/Schedule.java]: right, i get it now

**Pull request merged**